### DISCUSSION TITLE
How do we specify the reason for closing an issue?

### DISCUSSION THREAD

**Discussion by gbakeman** [community/community#56509] [Projects and Issues] [answered]
### Select Topic Area

Question

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

Related: #4514, #52048

I'm a little bit confused about the current behavior of issues insofar as closing them is concerned, per https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/.

![Issue close menu](https://i0.wp.com/user-images.githubusercontent.com/7584089/169088905-112804b4-ad7f-4bf0-b4f9-0b7d01e6a18e.png?w=512)

So we can close as `completed` or `not planned`, but what about the other options such as `duplicate`, `won't fix`, `stale`, etc.? I've already closed a few issues as not planned before when I want it to be duplicate, but I see no way to modify the reason.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by jge162** [comment]
You can always put notes when you close the issue as to why. But, you are correct with the limited options. Unless u use a GitHub action to close for example stale.

**Comment by orvn** [comment, marked as answer]
The `state_reason` key is currently in beta and isn't available just yet via the API or GraphQL endpoints, so I think this feature isn't up yet.

It's unclear whether or not this will be available via the Github UI, although it makes sense that it would be, and presumably it's blocked until the full release. The language from [the post](https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/#%F0%9F%95%B5%F0%9F%8F%BD%E2%99%80%EF%B8%8F-issue-closed-reasons) is a little misleading:

> We announced a preview of issue closed reasons in [March](https://github.blog/changelog/2022-03-10-the-new-github-issues-march-10th-update/) and today, we're releasing them to everyone.

That's not really true, since later on they specify that these are in preview and will be released later on.

**Reply by gbakeman** [reply]
Ok, it seemed as much. I think I saw a little confusion in the original discussion about how the feature worked. Are you aware of any issues currently tracking enabling and adding reason specification into the GitHub UI?

**Reply by orvn** [reply]
I haven't seen anything public, no
